src/recap.c

```diff
@@ -551,15 +551,19 @@ void eloop_tout_header_handler(void *eloop_ctx, void *user_ctx) {
   struct recap_context *pctx = (struct recap_context *)user_ctx;
 
   if (is_packet_queue_empty(pctx->pq) < 1) {
-    log_trace("Commiting packets to %s database", pctx->out_path);
-    if (execute_sqlite_query(pctx->db, "BEGIN IMMEDIATE TRANSACTION") < 0) {
-      log_error("Failed to capture a lock on db %s, ignoring.", pctx->out_path);
+    if (!pctx->pipe) {
+      log_trace("Commiting packets to %s database", pctx->out_path);
+      if (execute_sqlite_query(pctx->db, "BEGIN IMMEDIATE TRANSACTION") < 0) {
+        log_error("Failed to capture a lock on db %s, ignoring.", pctx->out_path);
```

Comment:
Should we make this a `log_warn` if we can safely ignore this?

```suggestion
        log_warn("Failed to capture a lock on db %s, ignoring.", pctx->out_path);
```